src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/FileKinds.cs

```diff
@@ -11,66 +9,67 @@ namespace Microsoft.AspNetCore.Razor.Language;
+    /// </summary>
+    public static bool IsComponentImport(this RazorFileKind fileKind)
+        => fileKind is RazorFileKind.ComponentImport;
 
-    public static bool IsComponent(string fileKind)
-    {
-        // fileKind might be null.
-        return string.Equals(fileKind, FileKinds.Component, StringComparison.OrdinalIgnoreCase) || IsComponentImport(fileKind);
-    }
+    /// <summary>
+    ///  Returns <see langword="true"/> if the specified value represents a legacy file kind.
+    /// </summary>
+    public static bool IsLegacy(this RazorFileKind fileKind)
+        => fileKind is RazorFileKind.Legacy;
 
-    public static bool IsComponentImport(string fileKind)
+    /// <summary>
+    ///  Compares the given file path to known Razor file extensions and names to determine the <see cref="RazorFileKind"/>.
+    ///  Returns <see langword="true"/> if a file kind can be determined; otherwise, <see langword="false"/>.
+    /// </summary>
+    public static bool TryGetFileKindFromPath(string filePath, out RazorFileKind fileKind)
     {
-        // fileKind might be null.
-        return string.Equals(fileKind, FileKinds.ComponentImport, StringComparison.OrdinalIgnoreCase);
-    }
+        ArgHelper.ThrowIfNull(filePath);
 
-#nullable enable
-    internal static bool IsLegacy(string? fileKind)
-    {
-        return string.Equals(fileKind, FileKinds.Legacy, StringComparison.OrdinalIgnoreCase);
-    }
-#nullable disable
+        var fileName = Path.GetFileName(filePath);
 
-    public static string GetComponentFileKindFromFilePath(string filePath)
-    {
-        if (filePath == null)
+        if (string.Equals(ComponentMetadata.ImportsFileName, fileName, StringComparison.Ordinal))
         {
-            throw new ArgumentNullException(nameof(filePath));
+            fileKind = RazorFileKind.ComponentImport;
+            return true;
         }
 
-        if (string.Equals(ComponentMetadata.ImportsFileName, Path.GetFileName(filePath), StringComparison.Ordinal))
-        {
-            return FileKinds.ComponentImport;
-        }
-        else
-        {
-            return FileKinds.Component;
-        }
-    }
+        var extension = Path.GetExtension(filePath);
 
-    public static string GetFileKindFromFilePath(string filePath)
-    {
-        if (filePath == null)
+        if (string.Equals(ComponentFileExtension, extension, StringComparison.OrdinalIgnoreCase))
         {
-            throw new ArgumentNullException(nameof(filePath));
+            fileKind = RazorFileKind.Component;
+            return true;
         }
 
-        if (string.Equals(ComponentMetadata.ImportsFileName, Path.GetFileName(filePath), StringComparison.Ordinal))
-        {
-            return FileKinds.ComponentImport;
-        }
-        else if (string.Equals(".razor", Path.GetExtension(filePath), StringComparison.OrdinalIgnoreCase))
+        if (string.Equals(LegacyFileExtension, extension, StringComparison.OrdinalIgnoreCase))
```

Comment:
I guess this is based on pre-existing code, but it's weird we use OrdinalIgnoreCase in most comparisons here and Ordinal when checking for _Imports.razor.

Reply:
Totally agree, but I took a look at the history and Ordinal has been used since that check was introduced.